lib/decode/decode.go

```diff
@@ -44,21 +44,28 @@ func HookTranslateKeys(_, to reflect.Type, data interface{}) (interface{}, error
 	}
 
 	rules := translationsForType(to)
+	// Avoid making a copy if there are no translation rules
+	if len(rules) == 0 {
+		return data, nil
+	}
+	result := make(map[string]interface{}, len(source))
 	for k, v := range source {
 		lowerK := strings.ToLower(k)
 		canonKey, ok := rules[lowerK]
 		if !ok {
+			result[lowerK] = v
```

Comment:
Should this use the original `k`? To avoid modifying existing keys that have no rules associated